src/record/record.js

```diff
@@ -297,12 +308,28 @@ Record.prototype.delete = function() {
  */
 Record.prototype.whenReady = function( callback ) {
 	if( this.isReady === true ) {
+		// TODO: What happens if callbacks makes calls on record?
 		callback( this );
 	} else {
+		// TODO: What happens if callbacks makes calls on record?
 		this.once( 'ready', callback.bind( this, this ) );
```

Comment:
Could be a problem if a callback calls discard before all callbacks have been called.